### ISSUE TITLE
Problem in dragonnet.py

### ISSUE THREAD

**Issue by matthewvowels1** [uber/causalml#600]
In the .fit() method, the code (and this occurs in the original version by Shi et al. as well) has a redundant optimization step. 

It firstly uses Adam, then,it uses SGD.

Checking with the original paper, the authos recommend the use of SGD, making the Adam step the one which ought to be removed.

Furthermore, the SGD step fails to take as argument the user specified learning rate and number of epochs, meaning that these are finally ignored during the optimization of the network.

Suggest these problems get fixed?

Cheers!
M

**Comment by matthewvowels1** [comment]
P.S. Having removed Adam I find the SGD training unstable when the targeted regularization is used. Thus maybe giving users the option of chosing between them would be the best option.

Note in either eventuality, the use specified epochs and learning rate should be passed to the respective optimizer (which they aren't currently).

Thanks!!
